design/milestone_1_2_design_doc.md

```diff
@@ -0,0 +1,455 @@
+
+   For a full list of required parameters, see `values.yaml` below
+
+   *When parameters are not configured by customer, their defaults are used*
+
+3. Install Helm Chart for the Secrets Provider
+
+We will supply the customer with a  `values.yaml` file where all default parameters are collected. Our `values.yaml` will resemble the following:
+
+```
+# K8s permissioning defaults
+permission:
+	serviceAccountName: secrets-provider-account
+	roleName: secrets-provider-role
+	roleBinding: secrets-provider-role-binding
+
+# Secrets Provider pod defaults
+secrets-provider:
+	image: cyberark/secrets-provider-for-k8s
+	name: cyberark-secrets-provider
+	namespace: default
+	ttlSecondsAfterFinished: 0	# Length at which Job should stay alive even after finishing task. Used for logging.
+	
+# Secrets Provider environment variable defaults
+secrets-provider-environment:
+	conjurApplianceUrl: https://conjur-follower.default.svc.cluster.local
+	conjurAuthnUrl: https://conjur-follower.default.svc.cluster.local/authn-k8s/authn-default
+	conjurAuthnLogin: host/conjur/authn-k8s/authn-default/default
+	conjurAccount: default
+	conjurSslCertificate: conjur-master-ca-env
+	debug=true
+	k8sSecrets=[]
+	containerMode=init
+	secretsDestination=k8s_secrets
+	rotationSupport=false
+```
+
+Customers can override these defaults by loading the Chart with `custom-values.yaml`. Helm will first look for values in the customer defined `custom-values.yaml`. If they do not exist, Helm will resort to the defaults configured in `values.yaml` and populate the manifest with those values. 
+
+For more detailed information on how a customer will install our Chart, see the section on [Installation](#installation).
+
+Variables from the `values.yaml` / `custom-values.yml `are applied for our project as follows:
+
+*templates/secrets-access-role.yaml*
+
+```
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ .Values.permission.serviceAccount }}
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: {{ .Values.permission.roleName }}
+rules:
+  - apiGroups: [""]
+    resources: {{ .Values.permission.rules.resources }}
+    verbs:
+    	rules:
+			resources: secrets
+			verbs: [ "get", "update" ]
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  namespace: {{ .Values.secrets-provider.namespace }}
+  name: {{ .Values.permission.roleBinding }}
+subjects:
+  - kind: ServiceAccount
+    namespace: {{ .Values.secrets-provider.namespace }}
+    name: {{ .Values.permission.serviceAccountName }}
+roleRef:
+  kind: Role
+  apiGroup: rbac.authorization.k8s.io
+  name: {{ .Values.permission.roleName }}
+```
+
+*templates/secrets-provider.yaml*
+
+The following template takes into account the anticipated flows for Milestone 1 and 2.
+
+```yaml
+---
+{{- if eq .Values.secrets-provider-environment.rotationSupport "false" }}
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ .Release.Name }}
```

Comment:
where does this come from?

Reply:
`.Release.Name` comes from the `Chart.yaml` so for our purposes
`name: secrets-provider`

Reply:
oh i see. thanks!